components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/TransactionHandlerComponent.java

```diff
@@ -0,0 +1,172 @@
+*
+* WSO2 Inc. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.micro.integrator.initializer.handler.transaction;
+
+import org.apache.axis2.transport.base.BaseConstants;
+import org.apache.axis2.util.JavaUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.config.mapper.ConfigParser;
+import org.wso2.micro.core.util.StringUtils;
+import org.wso2.micro.integrator.initializer.handler.transaction.security.CryptoUtil;
+import org.wso2.micro.integrator.initializer.handler.transaction.store.TransactionStore;
+import org.wso2.micro.integrator.ndatasource.common.DataSourceException;
+import org.wso2.micro.integrator.ndatasource.core.CarbonDataSource;
+import org.wso2.micro.integrator.ndatasource.core.DataSourceService;
+
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.crypto.Cipher;
+import javax.sql.DataSource;
+
+/**
+ * This class represents the internal transaction count handler component.
+ */
+public class TransactionHandlerComponent {
+
+    private static final Log LOG = LogFactory.getLog(TransactionHandlerComponent.class);
+    private static final int DEFAULT_SCHEDULED_PERIOD_INTERVAL = 60;
+    private static final String SCHEDULED_PERIOD = "transactionCountScheduledPeriod";
+
+    private static final String TRANSACTION_DATA_SOURCE = "transactionDataSource";
+    private static final String TRANSACTION_CONFIG_SECTION = "transaction";
+    private static final String TRANSACTION_CONFIG_NAME = "data_source";
+
+    public void start(DataSourceService dataSourceService) {
+        String nodeId = generateRandomId();
+        try {
+            // initialize cipher for encryption needs.
+            Cipher cipher = CryptoUtil.initializeCipher();
+
+            DataSource dataSource = getTransactionDataSource(dataSourceService);
+
+            // initialize Transaction store.
+            TransactionStore transactionStore = new TransactionStore(dataSource, nodeId, cipher);
+
+            // start and schedule transaction writer task.
+            scheduleTransactionWriterTask(transactionStore);
+
+        } catch (DataSourceException | TransactionException e) {
+            LOG.error("Error in initializing Transaction Service component: " + e.getMessage(), e);
```

Comment:
Shall we throw this and stop the server

Reply:
Even though this feature is enabled by default, we are not strictly enforcing the users to have this feature enabled. So IMO, warning or an error log would be enough to notify the users who do not want this feature and forget to disable it (or worst case, want the feature but forget to configure a dataSource). WDYT?

Reply:
I think if it is enabled by default , we will be having some storage by default and under normal circumstances this would work flawlessly. If the user edit some configs and having in mind that the transactions will be counted, IMO opinion it would be not that correct to print an error and proceed and later they figure out that their transaction count is missing which is irrecoverable.

Reply:
I think we should log an error and continue, if a datasource is not found. If it is located, we need to stop the server for all the subsequent errors

Reply:
Stopping the server if there are any DataSourceExceptions occurred while initializing the Transaction Count Handler component. Printing an error message otherwise.